Elements/src/Model.cs

```diff
@@ -272,8 +304,17 @@ private List<Element> RecursiveGatherSubElements(object obj)
                 return elements;
             }
 
-            var props = t.GetProperties(BindingFlags.Public | BindingFlags.Instance).Where(p => !p.GetCustomAttributes<JsonIgnoreAttribute>().Any());
-            var constrainedProps = props.Where(p => IsValidForRecursiveAddition(p.PropertyType));
+            List<PropertyInfo> constrainedProps;
+            if (properties.ContainsKey(t))
+            {
+                constrainedProps = properties[t];
+            }
+            else
+            {
+                constrainedProps = t.GetProperties(BindingFlags.Public | BindingFlags.Instance).Where(p => IsValidForRecursiveAddition(p.PropertyType) && p.GetCustomAttribute<JsonIgnoreAttribute>() != null).ToList();
```

Comment:
This query had a nice little speed boost when we filtered for valid types first _then_ filtered for custom attributes.